src/newrelic/processor_test.go

```diff
@@ -1139,3 +1158,155 @@ func TestShouldConnect(t *testing.T) {
+	if int(expectedLimit) != finalLimit {
+		t.Errorf("\" %s \": Expected %d to be choosen but %d was instead", testName, expectedLimit, finalLimit)
+	}
+
+	m.p.quit()
+}
+
+func TestConnectNegotiateLogEventLimits(t *testing.T) {
+
+	runMockedCollectorHarvestLimitTest(t, "log_event_data", 90*12, 100, "agent limit smaller than collector")
+	runMockedCollectorHarvestLimitTest(t, "log_event_data", 110*12, 100, "agent limit larger than collector")
+	runMockedCollectorHarvestLimitTest(t, "log_event_data", 100*12, 100, "agent limit equal to collector")
+	runMockedCollectorHarvestLimitTest(t, "log_event_data", 0, 100, "agent limit == 0, collector != 0")
+	runMockedCollectorHarvestLimitTest(t, "log_event_data", 100*12, 0, "agent limit != 0, collector == 0")
+
+}
+
+func TestProcessLogEventLimit(t *testing.T) {
+	//cannedConnectReply := []byte(`{"agent_run_id":"1"}`) // parseConnectReply expects at least agent_run_id in collector reply
+
+	// nil as argument should just return
+	processLogEventLimits(nil)
+
+	// empty App should just return (no info or connectReply)
+	emptyApp := App{}
+	processLogEventLimits(&emptyApp)
+
+	// only info defined should just return (no connectReply)
+	infoOnlyApp := App{info: &AppInfo{}}
+	processLogEventLimits(&infoOnlyApp)
+
+	// only connectReply defined should just return (no info)
+	connectReplyOnlyApp := App{connectReply: &ConnectReply{}}
+	processLogEventLimits(&connectReplyOnlyApp)
+}
+
+func TestMissingAgentAndCollectorHarvestLimit(t *testing.T) {
+
+	// tests missing agent and collector harvest limits will not crash daemon
+	// and results in empty final harvest values
+	appInfo := sampleAppInfo
+
+	m := NewMockedProcessor(1)
+
+	id := AgentRunID("log")
+
+	reply := m.p.IncomingAppInfo(nil, &appInfo)
+	<-m.p.trackProgress // receive app info
+	if reply.State != AppStateUnknown {
+		t.Fatal("Expected ", AppStateUnknown, " got ", reply.State)
+	}
+
+	// mock collector response specifying no harvest limit and test
+	// code handles it gravefully
+	mockedReply := `{"agent_run_id":"` + id.String() + `","zip":"zap"}`
+	m.DoConnectConfiguredReply(t, mockedReply)
+
+	reply = m.p.IncomingAppInfo(nil, &appInfo)
+	<-m.p.trackProgress // receive app info
+	if reply.State != AppStateConnected {
+		t.Fatal("Expected ", AppStateConnected, " got ", reply.State)
+	}
+
+	// should be 1 app in processor map and name should match that in appinfo
+	if 1 != len(m.p.apps) {
+		t.Error("Exactly one application should be created.")
+	}
+
+	app, found := m.p.apps[appInfo.Key()]
+	if !found {
+		t.Error("Expected application not found in processor map.")
+	}
+
+	expected := collector.EventHarvestConfig{}
+
+	if expected != app.connectReply.EventHarvestConfig {
+		actualStr, _ := json.MarshalIndent(app.connectReply.EventHarvestConfig, "", "\t")
+		expectedStr, _ := json.MarshalIndent(expected, "", "\t")
+		t.Errorf("Expected %s got %s", string(expectedStr), string(actualStr))
```

Comment:
Golang documentation has an opinionated and documented view on how test failure messages should be written [here (Useful Test Failures)](https://github.com/golang/go/wiki/CodeReviewComments#useful-test-failures) and [here (Got before Want)](https://github.com/golang/go/wiki/TestComments#got-before-want). Should this this simple convention be followed in newrelic-php-agent codebase?

Reply:
Great catch on this!

Looks like our current code base does both, as evidenced by this one file:
```
src/newrelic/processor_test.go
65:	sampleErrorEvent  = []byte("forgotten birthday")
236:		t.Fatalf("expected: %s \ngot: %s", expected, string(cp.data))
261:		t.Fatalf("expected: %s \ngot: %s", expected, string(cp.data))
290:		t.Fatalf("expected: %s \ngot: %s", expected, string(cp.data))
375:	if got := string(json); got != expectedJSON {
376:		t.Errorf("\ngot=%q \nwant=%q", got, expectedJSON)
398:	if string(cp.data) != `["one",{"reservoir_size":5,"events_seen":1},[forgotten birthday]]` {
624:		t.Fatal("Payload with 2500 events expected, got ", cp1Events)
627:		t.Fatal("Payload with 2500 events expected, got ", cp2Events)
644:		t.Fatal("Payloads with 4000 events expected, got ", cp1Events, " and ", cp2Events)
647:		t.Fatal("Payload sum of 8001 events expected, got ", cp1Events, " and ", cp2Events)
```

This is a good idea to standardize.  Our style guide can be updated with a `go` section to incorporate this guidance and the `go fmt` guidance.

Reply:
Im inclined to follow the convention of the file until we go through and change them everywhere otherwise the output is just going to be confusing to read.

Reply:
@mfulb I'm ok with your decision. I just wanted to bring to our attention that Golang has this convention.